cmd/upgrade/roles/cmd.go

```diff
@@ -0,0 +1,1012 @@
+				err = upgradeAccountRolePoliciesFromCluster(
+					mode,
+					reporter,
+					awsClient,
+					cluster,
+					creator.AccountID,
+					accountRolePolicies,
+					policyVersion,
+				)
+				if err != nil {
+					LogError(roles.RosaUpgradeAccRolesModeAuto, ocmClient, policyVersion, err, reporter)
+					if args.isInvokedFromClusterUpgrade {
+						return err
+					}
+					reporter.Errorf("Error upgrading the role polices: %s", err)
+					os.Exit(1)
+				}
+			}
+		case aws.ModeManual:
+			err = aws.GeneratePolicyFiles(reporter, env, isUpgradeNeedForAccountRolePolicies,
+				false, accountRolePolicies, nil)
+			if err != nil {
+				reporter.Errorf("There was an error generating the policy files: %s", err)
+				os.Exit(1)
+			}
+			if reporter.IsTerminal() {
+				reporter.Infof("All policy files saved to the current directory")
+				reporter.Infof("Run the following commands to upgrade the account role policies:\n")
+			}
+
+			commands, err := buildAccountRoleCommandsFromCluster(
+				mode,
+				cluster,
+				creator.AccountID,
+				isUpgradeNeedForAccountRolePolicies,
+				awsClient,
+				policyVersion,
+			)
+			if err != nil {
+				return err
+			}
+
+			fmt.Println(commands)
+			if args.isInvokedFromClusterUpgrade {
+				reporter.Infof("Run the following command to continue scheduling cluster upgrade"+
+					" once account and operator roles have been upgraded : \n\n"+
+					"\trosa upgrade cluster --cluster %s\n", r.ClusterKey)
+				return nil
+			}
+
+		default:
+			reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
+			os.Exit(1)
+		}
+	}
+
+	//OPERATOR ROLES
+	if !isInvokedFromClusterUpgrade {
+		reporter.Infof("Ensuring operator role policies compatibility for upgrade")
+	}
+
+	if spin != nil {
+		spin.Start()
+	}
+
+	operatorRoles, hasOperatorRoles := cluster.AWS().STS().GetOperatorIAMRoles()
+	if !hasOperatorRoles || len(operatorRoles) == 0 {
+		r.Reporter.Errorf("Cluster '%s' doesnt have any operator roles associated with it",
+			r.ClusterKey)
+	}
+
+	operatorRolePath, operatorPolicyPath, err := roles.GetOperatorPaths(r.AWSClient, operatorRoles)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	credRequests, err := r.OCMClient.GetCredRequests(cluster.Hypershift().Enabled())
+	if err != nil {
+		r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
+		os.Exit(1)
+	}
+
+	operatorRolePolicyPrefix, err := aws.GetOperatorRolePolicyPrefixFromCluster(cluster, r.AWSClient)
+	if err != nil {
+		return err
+	}
+
+	isOperatorPolicyUpgradeNeeded := false
+	isOperatorPolicyUpgradeNeeded, err = r.AWSClient.IsUpgradedNeededForOperatorRolePoliciesUsingCluster(
+		cluster,
+		r.Creator.AccountID,
+		policyVersion,
+		credRequests,
+		operatorRolePolicyPrefix,
+	)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	version := args.upgradeVersion
+
+	missingRolesInCS, err := r.OCMClient.FindMissingOperatorRolesForUpgrade(cluster, version)
```

Comment:
I see that `r.OCMClient` is assigned to a local variable in line `84`.
Should we use here `ocmClient` instead? or drop at once the local variables that are declared in lines 82-84`?